GV-144 Adding test gyms for 2 QA workflows tests (#6)
spatialos/UnrealGDKTestGyms: USER_MANUAL.md

  2. Additionally a handover value is incremented to monitor how many times this is called.
  3. The stack count, and the handover counter are then checked they are the same.

> #### QA Workflows Testing

Reviewer: Do we want to have a completely different section with different gyms? Can these gyms not be described the same way as the other ones and then workflow testing just says to look at those gyms?

Author: I can totally do that, I thought it might be better to keep separate, but happy to just include them without separate header, as I've already got reference to them in the test cases

##### ReplicatedStartupActors gym

> * For QA workflows Test Replicated startup actors are correctly spawned on all clients 

Reviewer: It looks like this is two sentences that have been merged somewhere.

* Validation

>   1. New Play Session is started 

Reviewer: I think this is implied in all the others.

  2. After two seconds checks that actor is visible to client and reports pass or fail

> ##### DestroyingActorsAtRuntime gym

Reviewer: If this is specifically startup actors maybe `DestroyStartupActorGym`?

Reviewer: In that case it can just destroy a startup actor and then connecting another client via any means can test that it's replicated correctly. But it keeps what the gym does simple and if you want to do stuff with clients you can. Otherwise I think we are venturing into what should be tested via other means.

* For QA workflows Test Demonstrates that when a Level Actor is destroyed by server, a late connecting client does not see this Actor
* Used to support QA test case "C1945 - Stably named actors can be destroyed at runtime and late-connecting clients don't see them"
* Validation:

>   1. New Play session is started, at 10 seconds Actor is deleted and success message is shown notifying the PIE clients

Reviewer: 10 seconds is quite a long time.

  2. Connect a client to Play session via command line
  ```
  sh

>   "C:\UnrealEngine\Engine\Binaries\Win64\UE4Editor.exe" "C:\UnrealGDKTestGyms\Game\GDKTestGyms.uproject" 127.0.0.1 -game -log -workerType UnrealClient -stdout -nowrite -unattended -nologtimes -nopause -noin -messaging -NoVerifyGC -windowed -ResX=800 -ResY=600

Reviewer: Absolute path in a shell script in a doc.

Author: I can remove this as it's listed in the test case, apologies but wasn't sure what level of information to include :)

  2. Depending on how the operations are scheduled, some clients/server workers will receive null references (red log message).
  3. Eventually, after one or more RepNotify, all workers should receive all the valid references (green log message).

##### ReplicatedStartupActor gym
* For QA workflows Test Replicated startup actor are correctly spawned on all clients
* Used to support QA test case "C1944 Replicated startup actors are correctly spawned on all clients" 
* Validation
  1. After two seconds checks that actor is visible to client and reports pass or fail

##### DestroyStartupActorGym gym
* For QA workflows Test Demonstrates that when a Level Actor is destroyed by server, a late connecting client does not see this Actor
* Used to support QA test case "C1945 - Stably named actors can be destroyed at runtime and late-connecting clients don't see them"
* Validation:
  1. At 10 seconds Actor is deleted and success message is shown notifying the PIE clients

>   2. In late connecting client Actor cannot be seen, when user presses `F` keyboard button, then searches for Actor and returns success or failure message 

Reviewer 2: ```suggestion
  2. Clients connecting after delete also cannot see Actor and on `F` keyboard button press searches for Actor and returns success or failure message 
```
